src/modules/components/proposal/proposalActions/actions/generators/proposalActionGeneric.ts

```diff
@@ -0,0 +1,12 @@
+import type { IProposalAction } from '../../proposalActionsTypes';
+
+export const generateProposalActionGeneric = (action?: Partial<IProposalAction>): IProposalAction => ({
```

Comment:
Usually these generators name matches the entity name, in this case I'd call this `generateProposalAction` and change the name as well